src/components/ui/button.tsx

```diff
@@ -0,0 +1,56 @@
+import { Slot } from "@radix-ui/react-slot"
+import { cva, type VariantProps } from "class-variance-authority"
+import * as React from "react"
+
+import { cn } from "~/lib/utils"
+
+const buttonVariants = cva(
+  "inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50",
```

Comment:
From a Chakra UI perspective, this right here is more than I can chew 😅 ... I thought I liked tailwind, but all these configurations are really making me reconsider.

Thanks for doing the hard work @iamstarcode.

Reply:
I didn't write all those in a day, 😂😂😂, you'd have to through shadcn docs, a simple line of code generated this, think it as a mini UI component library, independent of other UI components

Reply:
Okay. That makes sense 😃 😅